src/plugins/shared-variables/slate/variables-plugin.tsx

```diff
@@ -3,75 +3,141 @@ import classNames from "classnames/dedupe";
 
 import {
   BaseElement, CustomEditor, CustomElement, Editor, kSlateVoidClass, registerElementComponent,
-  RenderElementProps, Transforms, useSelected, useSerializing,
+  RenderElementProps, useSelected, useSerializing
 } from "@concord-consortium/slate-editor";
+import { action, autorun, computed, IReactionDisposer, makeObservable, observable } from "mobx";
+import { getType } from "mobx-state-tree";
+import { observer } from "mobx-react";
 import { VariableChip, VariableType } from "@concord-consortium/diagram-view";
-import { getVariables } from "./variables-text-content";
 import { TextContentModelType } from "../../../models/tiles/text/text-content";
-import { TextContentModelContext } from "../../../models/tiles/text/text-content-context";
+import { ITextPlugin } from "../../../models/tiles/text/text-plugin-info";
+import { TextPluginsContext } from "../../../components/tiles/text/text-plugins-context";
+
+import { DEBUG_SHARED_MODELS } from "../../../lib/debug";
+import { SharedVariables, SharedVariablesType } from "../shared-variables";
 
 const kVariableClass = "slate-variable-chip";
 export const kVariableFormat = "m2s-variable";
+export const kVariableTextPluginName = "variables";
+
+export class VariablesPlugin implements ITextPlugin{
+  public textContent;
+  private disposeSharedModelManagerAutorun: IReactionDisposer|undefined;
+
+  constructor(textContent: TextContentModelType) {
+    makeObservable(this, {
+      textContent: observable,
+      sharedModel: computed,
+      variables: computed,
+      onInitEditor: action
+    });
```

Comment:
I'm fine with this, but I prefer the decorator approach because it keeps the specification closer to the declaration where it seems less likely to be forgotten, and now that decorators are back on a standards track I'm not worried about support for them going away.

Reply:
I'm still not sure the right approach to decorators. The latest thing I'm finding is here: https://github.com/mobxjs/mobx/discussions/3373

So the new standardized decorators will eventually be supported, but it sounds like they'll be different than the current MobX 4/5 decorators. So I'm not sure if it is good idea to use the old ones, it might make it harder to upgrade to the new ones?

Reply:
I haven't been following it that closely so I may be wrong, but my sense is that the basic decorator syntax (`@observable`, `@computed`, `@action`) is likely to stay but that whether something like `makeObservable()` is necessary or some of the other edge cases are where things are likely to change. So the question for the near-term is weighing the relative risk of having to update the code whenever MobX adopts the standard decorators vs. the risk of inadvertently forgetting to declare something in the explicit `makeObservable()` syntax. Personally, I'm inclined to accept some of the former risk to mitigate the latter, but I can certainly see your argument. 🤷